README.md

```diff
@@ -4,227 +4,218 @@
+Rake tasks. The [Pisoni](https://github.com/3scale/pisoni) API client can be
+used to interact with the Internal API.
 
-We recommend editing code and committing locally, and executing tests within the
-container, since the container won't have your own tools and configurations.
+To quickly test Apisonator, random services can be created and configured on it
+via the use of the 'Buddhi' tool located in our performance tests
+toolkit: [3scale perftest-toolkit](https://github.com/3scale/perftest-toolkit/).
 
-### On local machine (unsupported)
+This component may also be referred to as '3scale_backend'.
 
-This is **unsupported** and **strongly discouraged** because it is the source of a lot
-of headaches and potential problems and introduces the need for us to document
-dependencies and config files here that are very likely to be outdated and
-incomplete.
+### Apisonator worker
 
-Bear in mind that we can only provide best effort support for this.
+It is responsible for performing background tasks off-loaded from
+Apisonator listeners (enqueued jobs).
 
-1. Install Redis v2.8.19 (or whichever version is recommended for use).
-2. Run `bundle install`.
-3. Configure Redis in `~/.3scale_backend.conf` as:
+The worker component takes care of running these enqueued jobs, mainly related
+to reporting of previous traffic.
 
-```
-ThreeScale::Backend.configure do |config|
-  config.redis.proxy = 'localhost:6379'
-end
-```
+Specifically, this component:
+ * Dequeues and runs the report jobs that have been submitted to the Redis
+   background jobs database by the Apisonator listener/s or the
+   Apisonator failed jobs rescheduler
+ * Stores the results of running the report jobs in the Redis Storage database
 
-## Running tests
+This component may also be referred to as '3scale_backend_worker'.
 
-You can either run them manually in the container-based development environment
-or have a container launched just for running the tests. For the latter you just
-need to run `make test`, and you can configure any additional environment
-variables for the test scripts like so:
+### Apisonator failed jobs rescheduler
 
-> `$ make DOCKER_OPTS="-e TEST_ALL_RUBIES=1" test`
+This is a simple task that acts as a cron scheduler to requeue jobs that failed
+when being processed by an Apisonator worker. The jobs are requeued into
+the Redis background jobs database.
 
-Another alternative to execute all tests from within the development environment
-where you should not need to manually start/stop the services before is to execute
-them from inside the development environment via this script:
+This component may also be referred to as 'backend-cron'.
 
-> `$ script/test`
+## Development
 
-### Running tests (advanced)
+See the file [DEVELOPMENT](DEVELOPMENT.md)
 
-In order to run the tests manually in the container-based development, the services
-needed to run them correctly must be started before with:
+## Deployment set up
```

Comment:
I would not call this section *deployment* 

This should be the tipical: "How to run".  Deployment is out of the scope of this document